### DISCUSSION TITLE
Duplicate File Names and Renaming files?

### DISCUSSION THREAD

**Discussion by techoguy** [outdoorbits/little-backup-box#333] [Q&A]
I found this great project and want to build one but I had a couple questions before I did. 

1. I have a Nikon which creates file names sequentially which means file names can be duplicated (between deleting files and the count cycles around), how are people handling this to make sure different pictures are NOT overwritten? Or is there a place to look to better understand how it backs up or decides which files are backed up? 
2. When backing up, is there a way to automatically change the file name to something like a date/time formatted name (Nikon's won't do this) based on file creation date or metadata in the picture? 
3. how do you deal with power outages? If the pi is plugged into the wall and you're transferring videos, large files, how do most people protect the target device from corruption etc? Or is it a risk without a pi UPS device to gracefully shut the pi down?

**Comment by outdoorbits** [comment]
1. My Nikon Z6 saves files in subfolders of DCIM like 103NCZ_6, 104NCZ_6 
and so on. This folder structure is kept in lbb's backup. That means, 
collisions are impossible. I'm not sure, but I would expect, any Nikon 
camera would work like this?

2. There is no renaming yet. It wouldn't be a very big deal to implement 
this after backup. Maybe a nice idea. I configured my Z6 to name the 
images like "Z6P_2309.NEF". Why not renaming it to "2024-10-18 11-26-51 
- Z6P_2309.NEF"? Is this what you mean?

**Reply by techoguy** [reply]
1) My file structure is the same (as long as you have the sequential file name setting on within the Nikon), but technically the folder auto numbering also reaches a limit before it has to be reset and folders and file names repeat. That would be a lot of photos of course and unlikely to happen. As for if you have multiple cameras, I'm assuming you have to use different prefixes for the file/folder names or make sure to use different SD cards all the time so you don't run into duplicates. I was not considering the idea that LBB copies the folder structure too, so I can see how this isn't much of a problem in practice. 

2) Correct, a way to rename photos is great because I'm hoping to work towards getting away from Nikon's file structure which is required because of the short names they use. Of course, if a picture is renamed, I wouldn't want it to re-download when the source device is plugged in again at a later time. Ideally, I get to a point where all photos can be in one folder ( or at least in a file structure I define, maybe based on month, year etc). 

3) I edited the original post to add 3...

**Comment by outdoorbits** [comment]
1: Backup directly from camera is done to a path containing the cameras 
serial. on backup from storage cards every card gets a unique ID. This 
ID is part of the backup path.

2: After implementing a renaming step: If you keep your files at 
source,they will be copied again next time, because the original file 
name does not exist at target. I do not see a clean and secure way to 
prevent this.

3: Powercuts are always a big risk to break linux, be sure to have one 
or two spare lbb-SDs with you in case one gets broken. Incomplete 
transferred files will be copied next time.

**Reply by techoguy** [reply]
2) makes sense, wasn't sure if copied files were tracked in a db such that there could be a translation between source and target file names. I'd still be interested in a feature to batch change names based on date take etc. 

3) okay so external drives should be okay during backup, but the OS drive is the risk...what about the case where an NVMe SSDs is used as the OS device and target? Do you run the risk of losing everything backed up on the SSD or more just the risk of not being able to boot up Linux? Meaning, is it saver to keep the OS and target separate? 

Thanks for all the quick replies!

**Comment by outdoorbits** [comment]
My decission is clear, I separate data and os. I do not expect to lose 
all the data in case of a crashed os but why to take a risk?

**Reply by techoguy** [reply]
That seems like a smart decision and agree with that. Thanks again for clarifying and quick responses!

**Comment by outdoorbits** [comment]
The renaming function is implemented, I just forgot to post this here, sorry.
